src/utils/getFlowType.js

```diff
@@ -82,6 +82,19 @@ function handleKeysHelper(path: NodePath): ?FlowElementsType {
         elements: keys.map(key => ({ name: 'literal', value: key })),
       };
     }
+  } else if (
+    resolvedPath &&
+    types.ObjectTypeAnnotation.check(resolvedPath.node) &&
+    resolvedPath.node.properties.length
+  ) {
+    return {
+      name: 'union',
+      raw: printValue(path),
+      elements: resolvedPath.get('properties').map(prop => ({
+        name: 'literal',
+        value: getPropertyName(prop),
+      })),
+    };
```

Comment:
I wonder if we can consolidate this even more since I guess nodes of type `ObjectExpression` and `ObjectTypeAnnotation` are very similar. At the very least, can we extract an array of key names for them and process that?